website/docs/language/import/index.mdx

```diff
@@ -0,0 +1,188 @@
+
+The `import` block has the following arguments:
+ - `to` - The resource instance address targeted by this import.
+ - `id` - The import ID of the resource. String.
+ - `provider` (optional) - A custom resource provider. See [The Resource provider Meta-Argument](/terraform/language/meta-arguments/resource-provider).
+
+## Planning and applying an import
+
+The `import` block, like all Terraform configuration blocks, is processed during the `terraform plan` operation, which outputs a list of changes Terraform will make if you proceed to apply. Along with create, update, remove, and move, import is an action Terraform can perform on a resource. The resource is only added to the state during `terraform apply`.
+
+### 1. Add the `import` block to configuration
+
+```hcl
+provider "aws" {
+  region = "eu-west-1"
+}
+
+import {
+  to = aws_iot_thing.bar
+  id = "foo"
+}
+```
+
+### 2. Add the `resource` block to configuration
+
+```hcl
+resource aws_iot_thing "bar" {
+  name = "foo"
+}
+```
+
+### 3. Plan the import
+Run `terraform plan`. You should see output like the following:
+
+
+```
+$ terraform plan
+
+aws_iot_thing.bar: Preparing import... [id=foo]
+aws_iot_thing.bar: Refreshing state... [id=foo]
+
+Terraform will perform the following actions:
+
+  # aws_iot_thing.bar will be imported
+    resource "aws_iot_thing" "bar" {
+        arn               = "arn:aws:iot:eu-west-1:1234567890:thing/foo"
+        attributes        = {}
+        default_client_id = "foo"
+        id                = "foo"
+        name              = "foo"
+        version           = 1
+    }
+
+Plan: 1 to import, 0 to add, 0 to change, 0 to destroy.
+```
+
+### 4. Apply the import
+Run `terraform apply`. You should see output like the following:
+
+```
+$ terraform apply
+
+aws_iot_thing.bar: Preparing import... [id=foo]
+aws_iot_thing.bar: Refreshing state... [id=foo]
+
+Terraform will perform the following actions:
+
+  # aws_iot_thing.bar will be imported
+    resource "aws_iot_thing" "bar" {
+        arn               = "arn:aws:iot:eu-west-1:1234567890:thing/foo"
+        attributes        = {}
+        default_client_id = "foo"
+        id                = "foo"
+        name              = "foo"
+        version           = 1
+    }
+
+Plan: 1 to import, 0 to add, 0 to change, 0 to destroy.
+aws_iot_thing.bar: Importing... [id=foo]
+aws_iot_thing.bar: Import complete [id=foo]
+
+Apply complete! Resources: 1 imported, 0 added, 0 changed, 0 destroyed.
+
+```
+
+## Resource configuration
+
+When Terraform imports a resource, it needs the resource to be defined in configuration before saving it to the state. This means that when using `import` blocks, you have two options.
+
+### Option: Write a `resource` block
+
+Write a `resource` block for the resource you are importing, whose address matches the import block target. For example:
```

Comment:
Can you explain why someone would do one or the other? It's helpful to add guidance for what we _recommend_ folks do.